template/data/src/main/java/co/nimblehq/template/data/repository/RepositoryImpl.kt

```diff
@@ -1,23 +1,17 @@
 package co.nimblehq.template.data.repository
 
+import co.nimblehq.template.data.extensions.flowTransform
 import co.nimblehq.template.data.response.toModels
 import co.nimblehq.template.data.service.ApiService
 import co.nimblehq.template.domain.model.Model
 import co.nimblehq.template.domain.repository.Repository
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flow
 
 class RepositoryImpl constructor(
     private val apiService: ApiService
 ) : Repository {
 
-    override fun getModels(): Flow<List<Model>> = flow {
-        try {
-            val result = apiService.getResponses().toModels()
-            emit(result)
-        } catch (exception: Exception) {
-            // TODO do error mapping here
-            throw exception
-        }
+    override fun getModels(): Flow<List<Model>> = flowTransform {
```

Comment:
From [this discussion](https://github.com/nimblehq/android-templates/pull/311/files#r971788418), why don't you omit the explicit type here since you have the `@BuilderInference` annotation for this?
```suggestion
    override fun getModels() = flowTransform {
```

Reply:
@lydiasama As I said here https://github.com/nimblehq/android-templates/pull/311/files#r971836394, it requires Java to be updated to 11, and it requires the CI to use Java 11 also https://github.com/nimblehq/android-templates/blob/develop/.github/workflows/review_pull_request.yml#L13-L16.

Not sure if we really need to update to use Java 11 in this PR. However, I think it's good to keep the support for the next Java version with `@BuilderInference`.

Reply:
@luongvo Updating Java version seems out of scope for this and could introduce a whole bunch of issues ([for example](https://github.com/nimblehq/android-templates/issues/163#issuecomment-1081553940)). Let's stick with this solution as is for now 😉